nexusadspy/segment.py

```diff
@@ -0,0 +1,136 @@
+
+from __future__ import (
+    print_function, division, generators,
+    absolute_import, unicode_literals
+)
+
+from nexusadspy.client import AppnexusClient
+from io import BytesIO
+from gzip import GzipFile
+
+
+class MemberOfSegment:
+    """
+    Class to hold attributes about the user to be added in segment.
+    """
+    def __init__(self,appnexus_user_id, timestamp, mobile_os=None, expiration=0, numeric_value=0):
+        """
+        Detailed documentation about values can be found here:
+        https://wiki.appnexus.com/display/api/Batch+Segment+Service+-+File+Format
+        :param appnexus_user_id: str, Appnexus ID of the user. IDFA/AAID in case of mobiles.
+        :param timestamp: int, Unix timestamp at which user goes into effect in segment.
+        :param mobile_os: str, mobile OS used by user. None if desktop.
+        :param expiration: int, Lifetime of user in segment (e.g. android, ios). 0 means never expire.
+        :param numeric_value: int, Numeric value for the user.
+        """
+        self._user_id = appnexus_user_id
+        self._expiration = expiration
+        self._timestamp = timestamp
+        self._numeric_value = numeric_value
+        self._mobile_os = mobile_os
+
+    def get_user_id(self):
+        return self._user_id
+
+    def get_timestamp(self):
+        return self._timestamp
+
+    def get_expiration(self):
+        return self._expiration
+
+    def get_value(self):
+        return self._numeric_value
+
+    def get_mobile_os(self):
+        return self._mobile_os
+
+
+class SegmentsUploaderJob:
+    """
+    batch-upload API wrapper for appnexus
+    """
+
+    BATCH_UPLOAD_ANDROID_SPECIFIER = "8"
+    BATCH_UPLOAD_IOS_SPECIFIER = "3"
+
+    def __init__(self, users_list, segment_code, separators, member_id,
+                 credentials_path='.appnexus_auth.json'):
+        """
+        Initialize the uploader
+        :param users_list: list, List of MemberOfSegment. Users to be added in segment.
+        :param segment_code: str, Segment code to add users in.
+        :param separators: array, Array of five field separators. As documented in 
+        https://wiki.appnexus.com/display/api/Batch+Segment+Service+-+File+Format#BatchSegmentService-FileFormat-Separators
+        :param member_id: str, Member ID for appnexus account.
+        :param credentials_path: str, Credentials path for AppnexusClient.
+        :return:
+        """
+        self._appnexus_client = AppnexusClient(credentials_path)
```

Comment:
Isn't authentication already lazy? `AppnexusClient` does not do authentication in constructor. It does it in `request`.
Right now I am initialising it in constructor because more than one class methods use it.